internal/translation/teams/teams.go

```diff
@@ -0,0 +1,130 @@
+package teams
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type TeamsService interface {
+	ListProjectTeams(ctx context.Context, projectID string) ([]AssignedTeam, error)
+	GetTeamByName(ctx context.Context, orgID, teamName string) (*AssignedTeam, error)
+	GetTeamByID(ctx context.Context, orgID, teamID string) (*AssignedTeam, error)
+	Assign(ctx context.Context, at *[]AssignedTeam, projectID string) error
+	Unassign(ctx context.Context, projectID, teamID string) error
+	Create(ctx context.Context, at *Team, orgID string) (*Team, error)
+	RenameTeam(ctx context.Context, at *AssignedTeam, orgID, newName string) (*AssignedTeam, error)
+	UpdateRoles(ctx context.Context, at *AssignedTeam, projectID string, newRoles []akov2.TeamRole) error
+
+	GetTeamUsers(ctx context.Context, orgID, teamID string) ([]TeamUser, error)
+	AddUsers(ctx context.Context, usersToAdd *[]admin.AddUserToTeam, orgID, teamID string) error
+	RemoveUser(ctx context.Context, orgID, teamID, userID string) error
+}
```

Comment:
Trying to order my head around these, it would seem we are actually wrapping 3 sub services.

type TeamsService interface {
  TeamProjectsService
  TeamRolesService
  TeamMembersService
}

type TeamProjectsService interface { // manages Team's associations to Projects
  	ListProjectTeams(ctx context.Context, projectID string) ([]TeamRoles, error)
	Assign(ctx context.Context, at *[]TeamRoles, projectID string) error
	Unassign(ctx context.Context, projectID, teamID string) error
}

type TeamRolesService interface { // manages Team's Roles
	GetTeamByName(ctx context.Context, orgID, teamName string) (*TeamRoles, error)
	GetTeamByID(ctx context.Context, orgID, teamID string) (*TeamRoles, error)
	RenameTeam(ctx context.Context, at *TeamRoles, orgID, newName string) (*TeamRoles, error)
	UpdateRoles(ctx context.Context, at *TeamRoles, projectID string, newRoles []akov2.TeamRole) error
}

Type TeamMembersService interface { // manages Team's Members (Users)
	GetTeamUsers(ctx context.Context, orgID, teamID string) ([]TeamUser, error)
	AddUsers(ctx context.Context, usersToAdd *[]admin.AddUserToTeam, orgID, teamID string) error
	RemoveUser(ctx context.Context, orgID, teamID, userID string) error
}

Reply:
Agreed, this is more clear. Will implement 👍